### PR TITLE
SONARHTML-215 Create rule S6793: ARIA properties in DOM elements should have valid values

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**

### CODE REVIEW

File: sonar-html-plugin/src/test/resources/checks/AriaProptypesCheck.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<!-- Non-aria attribute -->
+<div aria-foo="true" />
+<div foo="true" />
+
+<!-- Boolean -->
+<div aria-hidden="true" />
+<div aria-hidden="false" />
+<div aria-hidden="undefined" />
```
Comment (ilia-kebets-sonarsource) [inline sonar-html-plugin/src/test/resources/checks/AriaProptypesCheck.html:8]: should we support `undefined` in HTML? I mean it makes sense in JSX, but here?

File: sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/Aria.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    ARIA_PROPERTIES.put("aria-valuemin", new AriaProperty("aria-valuemin", AriaPropertyType.NUMBER));
+    ARIA_PROPERTIES.put("aria-valuenow", new AriaProperty("aria-valuenow", AriaPropertyType.NUMBER));
+    ARIA_PROPERTIES.put("aria-valuetext", new AriaProperty("aria-valuetext", AriaPropertyType.STRING));
+  }
+
+  static class AriaProperty {
+    private final String name;
+    private final AriaPropertyType type;
+    private final Optional<Boolean> allowUndefined;
+    private final List<String> values;
```
Comment (ilia-kebets-sonarsource) [inline sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/Aria.java:88]: why not use a Set?

File: sonar-html-plugin/src/test/resources/checks/AriaProptypesCheck.html (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply sonar-html-plugin/src/test/resources/checks/AriaProptypesCheck.html:8]: Indeed, it doesn't make any sense here.

File: sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/Aria.java (hunk 2 above)
Comment (yassin-kammoun-sonarsource) [reply sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/Aria.java:88]: Sure, but I won't be able to assert some issue messages based on this set as the value order won't be deterministic on collection.

Comment (ilia-kebets-sonarsource) [reply sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/Aria.java]: You did the change, meaning it's all good?

File: sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/AriaProptypesCheck.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+      var value = attribute.getValue();
+      var property = ARIA_PROPERTIES.get(normalizedName);
+      if (!isValid(property, value)) {
+        createViolation(element.getStartLinePosition(), message(name, property.getType(), property.getValues()));
+      }
+    }
+  }
+
+  private static boolean isValid(AriaProperty property, String value) {
```
Comment (ilia-kebets-sonarsource) [inline sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/AriaProptypesCheck.java:54]: These might be candidates to be moved to the `Aria.java` helper class.

File: sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/Aria.java (hunk 2 above)
Comment (yassin-kammoun-sonarsource) [reply sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/Aria.java]: Yes

File: sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/AriaProptypesCheck.java (hunk 3 above)
Comment (yassin-kammoun-sonarsource) [reply sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/AriaProptypesCheck.java:54]: I don't think we need to. This is really specific to the rule, and the JavaScript implementation doesn't expose it as a helper function anywhere outside the rule.

**Pull request merged**